### PR TITLE
Throw InvalidStateError when calling play() on a finished, reversed, infinite duration animation

### PR DESCRIPTION
**Pull request by alancutter**
Relevant spec section:
https://w3c.github.io/web-animations/#playing-an-animation-section

### CODE REVIEW

File: src/animation.js [hunk 1]
```diff
@@ -141,7 +144,15 @@
     play: function() {
       this._paused = false;
       if (this._isFinished || this._idle) {
-        this._currentTime = this._playbackRate > 0 ? 0 : this._totalDuration;
+        if (this._playbackRate >= 0) {
```
Comment (suzyh) [inline src/animation.js]: Is it significant that this has gone from > 0 to >= 0?

Comment (alancutter) [inline src/animation.js]: It kind of emulates the part in the spec that says "Set animation’s hold time to zero." when we don't really have the concept of a hold time here.

**Comment by suzyh** [comment]
Given that this has caused the finished play state tests to go backwards, I'm reluctant to merge this until it's clear that that can be fixed.

**Comment by alancutter** [comment]
finish() was already broken.
See follow up patch to make finish() a bit less broken: https://github.com/web-animations/web-animations-next/issues/477

**Comment by suzyh** [comment]
ack, lgtm

**Pull request merged**